From e6369b8986953e3d0eba93cc024a2219320fa42c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Tue, 22 Oct 2019 14:52:17 +0200
Subject: [PATCH 1/2] Revert "FBA DASD should use the msdos disk label type"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Organization: Invisible Things Lab
Cc: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>

This reverts commit 41627e946cafc9341fd90c6713830aa67ebe28e7.

Signed-off-by: Marek Marczykowski-Górecki <marmarek@invisiblethingslab.com>
---
 blivet/platform.py | 21 ++++-----------------
 python-blivet.spec |  2 +-
 2 files changed, 5 insertions(+), 18 deletions(-)

diff --git a/blivet/platform.py b/blivet/platform.py
index 4772b35d..9100487e 100644
--- a/blivet/platform.py
+++ b/blivet/platform.py
@@ -22,11 +22,6 @@
 import logging
 log = logging.getLogger("blivet")
 
-import gi
-gi.require_version("BlockDev", "2.0")
-
-from gi.repository import BlockDev as blockdev
-
 import parted
 
 from . import arch
@@ -356,20 +351,12 @@ class S390(Platform):
         return [PartSpec(mountpoint="/boot", size=Size("1GiB"),
                          weight=self.weight(mountpoint="/boot"), lv=False)]
 
-    def best_disklabel_type(self, device):
-        """The best disklabel type for the specified device."""
-        if flags.testing:
-            return self.default_disklabel_type
-
-        # the device is FBA DASD
-        if blockdev.s390.dasd_is_fba(device.path):
-            return "msdos"
-        # the device is DASD
-        elif parted.Device(path=device.path).type == parted.DEVICE_DASD:
+    def required_disklabel_type(self, device_type):
+        """The required disklabel type for the specified device type."""
+        if device_type == parted.DEVICE_DASD:
             return "dasd"
 
-        # other types of devices
-        return super(S390, self).best_disklabel_type(device)
+        return super(S390, self).required_disklabel_type(device_type)
 
 
 class ARM(Platform):
diff --git a/python-blivet.spec b/python-blivet.spec
index 1481bf84..0c3392c9 100644
--- a/python-blivet.spec
+++ b/python-blivet.spec
@@ -19,7 +19,7 @@ Source0: http://github.com/storaged-project/blivet/archive/%{realname}-%{realver
 %global partedver 1.8.1
 %global pypartedver 3.10.4
 %global utillinuxver 2.15.1
-%global libblockdevver 2.6
+%global libblockdevver 2.1
 %global libbytesizever 0.3
 %global pyudevver 0.18
 
-- 
2.20.1

